pkg/providers/ingress/translation/annotations/plugins/http_method.go

```diff
@@ -0,0 +1,83 @@
+// Licensed to the Apache Software Foundation (ASF) under one or more
+// contributor license agreements.  See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package plugins
+
+import (
+	"github.com/apache/apisix-ingress-controller/pkg/providers/ingress/translation/annotations"
+	apisixv1 "github.com/apache/apisix-ingress-controller/pkg/types/apisix/v1"
+)
+
+type HttpMethod struct{}
+
+// NewHttpHandler creates a handler to convert annotations about
+// HttpMethod to APISIX cors plugin.
+func NewHttpMethodHandler() PluginAnnotationsHandler {
+	return &HttpMethod{}
+}
+
+func (h HttpMethod) PluginName() string {
+	return "response-rewrite"
+}
+
+func (h HttpMethod) Handle(e annotations.Extractor) (interface{}, error) {
+	var plugin apisixv1.ResponseRewriteConfig
+
+	allowMethods := e.GetStringsAnnotation(annotations.AnnotationsHttpAllowMethod)
+	blockMethods := e.GetStringsAnnotation(annotations.AnnotationsHttpBlockMethod)
+
+	plugin.StatusCode = 405
+	var methodExpr = []apisixv1.Expr{}
+
+	if len(allowMethods) > 0 {
+		//plugin.LuaRestyExpr = []interface{}{
+		//	[]interface{}{
+		//		"request_method", "!", "in", allowMethods,
+		//	},
+		//}
+		for _, method := range allowMethods {
+			methodExpr = append(methodExpr, apisixv1.Expr{StringVal: method})
+		}
+
+		plugin.LuaRestyExpr = []apisixv1.Expr{
+			{
+				ArrayVal: []apisixv1.Expr{
+					{StringVal: "request_method"},
+					{StringVal: "!"},
+					{StringVal: "in"},
+					{ArrayVal: methodExpr},
+				},
+			},
+		}
+
+	} else if len(blockMethods) > 0 {
```

Comment:
This means that the two annotations are mutually exclusive. We should make it clear in the docs, thanks